tools/devops/device-tests/templates/device-tests.yml

```
@@ -251,9 +251,14 @@ steps:
     cd $WORKING_DIR/xamarin-macios
     if [[ "$USE_XAMARIN_STORAGE" == "True" ]]; then
       if nc -z xamarin-storage 22 2>/dev/null; then
-        ssh builder@xamarin-storage "mkdir -p /volume1/storage/$XAMARIN_STORAGE_PATH"
-        echo '##vso[task.setvariable variable=XAMARIN_STORAGE_FAILED;isOutput=true]false'
-        echo '##vso[task.setvariable variable=XAMARIN_STORAGE_PATH;isOutput=true]$XAMARIN_STORAGE_PATH'
+        EC=0
+        MKSTORAGE="ssh builder@xamarin-storage "mkdir -p /volume1/storage/$XAMARIN_STORAGE_PATH""
+        eval $MKSTORAGE || EC=$?
+        if [ $EC -eq 0 ]; then
+          echo '##vso[task.setvariable variable=XAMARIN_STORAGE_PATH;isOutput=true]$XAMARIN_STORAGE_PATH'
```

You forgot to add the variable XAMARIN_STORAGE_FAILED to be false, better to add it in case we have to check the value. Also we need to move the export to inside the if.
```suggestion
          echo '##vso[task.setvariable variable=XAMARIN_STORAGE_PATH;isOutput=true]$XAMARIN_STORAGE_PATH'
          echo '##vso[task.setvariable variable=XAMARIN_STORAGE_FAILED;isOutput=true]false'
          export TESTS_PERIODIC_COMMAND="--periodic-interval 10 --periodic-command rsync --periodic-command-arguments '-avz -e \"ssh\" $PWD/jenkins-results builder@xamarin-storage:/volume1/storage/$XAMARIN_STORAGE_PATH'"
```